cura/Machines/Models/NozzleModel.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) 2018 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+
+from PyQt5.QtCore import Qt
+
+from UM.Application import Application
+from UM.Qt.ListModel import ListModel
+from UM.Util import parseBool
+
+
+class NozzleModel(ListModel):
+    IdRole = Qt.UserRole + 1
+    HotendNameRole = Qt.UserRole + 2
+    ContainerNodeRole = Qt.UserRole + 3
+
+    def __init__(self, parent = None):
+        super().__init__(parent)
+
+        self.addRoleName(self.IdRole, "id")
+        self.addRoleName(self.HotendNameRole, "hotend_name")
+        self.addRoleName(self.ContainerNodeRole, "container_node")
+
+        Application.getInstance().globalContainerStackChanged.connect(self._update)
+        Application.getInstance().getMachineManager().activeVariantChanged.connect(self._update)
+        Application.getInstance().getMachineManager().activeStackChanged.connect(self._update)
+        Application.getInstance().getMachineManager().activeMaterialChanged.connect(self._update)
```

Comment:
I think this can be simplified, but we have tested this and it worked. So, we decide to leave them as it is right now.

Reply:
Well yes, it'll work, but it will be three times as slow as it should be. Right now that may not yet be a problem, but as Cura gets developed this usually finds its way back as the next "interface is slow to react" type of bug.

Reply:
I've made this more efficient here: 00783401752a3bbc661fc6d65c15f6fd7861bdee
And the material switching here: 5afcf2beacfac24b56d11bab0da4a3f5efe6b1bf

I'm still getting the `globalContainerStackChanged` signal multiple times though, quite often. Can't listen more granular there, I'm afraid.